src/ArduinoIoTCloudTCP.cpp

```diff
@@ -270,6 +270,12 @@ void ArduinoIoTCloudTCP::update()
   }
   _state = next_state;
 
+#if OTA_ENABLED
+  if(_state >= State::SubscribeDeviceTopic && _state <= State::Connected) {
```

Comment:
```suggestion
  if(_state > State::SubscribeDeviceTopic && _state <= State::Connected) {
```
If `State` is exactly `SubscribeDeviceTopic` you cannot be sure that the board is correctly subscribed to the device topic